components/forms/CreateTxForm/MsgForm/MsgTransferForm.tsx

```diff
@@ -0,0 +1,233 @@
+}
+
+const MsgTransferForm = ({ fromAddress, setMsgGetter, deleteMsg }: MsgTransferFormProps) => {
+  const { state } = useAppContext();
+  assert(state.chain.addressPrefix, "addressPrefix missing");
+
+  const [sourcePort, setSourcePort] = useState("transfer");
+  const [sourceChannel, setSourceChannel] = useState("");
+  const [denom, setDenom] = useState("");
+  const [amount, setAmount] = useState("0");
+  const [toAddress, setToAddress] = useState("");
+  const [timeout, setTimeout] = useState("");
+  const [memo, setMemo] = useState("");
+
+  const [sourcePortError, setSourcePortError] = useState("");
+  const [sourceChannelError, setSourceChannelError] = useState("");
+  const [denomError, setDenomError] = useState("");
+  const [amountError, setAmountError] = useState("");
+  const [toAddressError, setToAddressError] = useState("");
+  const [timeoutError, setTimeoutError] = useState("");
+
+  useEffect(() => {
+    setSourcePortError("");
+    setSourceChannelError("");
+    setDenomError("");
+    setAmountError("");
+    setToAddressError("");
+    setTimeoutError("");
+
+    const isMsgValid = (msg: TxMsg): msg is TxMsgTransfer => {
+      if (!sourcePort) {
+        setSourcePortError("Source port is required");
+        return false;
+      }
+
+      if (!sourceChannel) {
+        setSourceChannelError("Source channel is required");
+        return false;
+      }
+
+      if (!denom) {
+        setDenomError("Denom is required");
+        return false;
+      }
+
+      if (!amount || Number(amount) <= 0) {
+        setAmountError("Amount must be greater than 0");
+        return false;
+      }
+
+      const addressErrorMsg = checkAddress(toAddress, ""); // Allow address from any chain
+      if (addressErrorMsg) {
+        setToAddressError(`Invalid address for network ${state.chain.chainId}: ${addressErrorMsg}`);
+        return false;
+      }
+
+      const foundTimeout = humanTimestampOptions.find(({ label }) => label === timeout);
+      const isTimeoutNumber = !isNaN(Number(timeout));
+      const isTimeoutInFuture = Number(timeout) > Date.now();
+
+      if (!foundTimeout || !isTimeoutNumber || !isTimeoutInFuture) {
+        setTimeoutError("Timeout must be a valid timestamp in the future");
+      }
+
+      return isTxMsgTransfer(msg);
+    };
+
+    const timeoutTimestamp = (() => {
+      const foundTimeout = humanTimestampOptions.find(({ label }) => label === timeout)?.value;
+      if (foundTimeout) {
+        return Long.fromNumber(Date.now() + foundTimeout).multiply(1_000_000); // In nanoseconds
+      }
+
+      try {
+        return Long.fromString(timeout);
+      } catch {
+        return Long.fromNumber(0);
+      }
+    })();
+
+    const msg: TxMsgTransfer = {
+      typeUrl: "/ibc.applications.transfer.v1.MsgTransfer",
+      value: {
+        sender: fromAddress,
+        receiver: toAddress,
+        token: { denom, amount },
+        sourcePort,
+        sourceChannel,
+        timeoutTimestamp,
```

Comment:
okay what I missed here is that we need to set `timeoutTimestamp` to `Math.floor(Date.now()/1000) + diffProvidedByUser`

Reply:
That is calculated for the dropdown menu, for the manual input the user needs to enter the final timestamp with the current logic

Reply:
Oh indeed, we have this logic above 👍